### ISSUE TITLE
Simple possible improvement to Comments field in dialogues

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#4825]
In preparing the guide to help those who may graduate to R it would be good if the Comments - that is on all dialogues could sometimes be more extensive.
In discussion with Danny we now suggest a small ... button which then provides a textbox into which the comment can be typed.  The default comment might be copied into it.  It would permit multiline comments.  It is sent to R just as the existing comment, but with a # in front of each line.

No special urgency, except it will affect all dialogues and hence would be good to do soon if we are going to do it.  I have therefore made it as 5.1.
If it is quite a bit of work, then I suggest we add the ... button very soon, (which is on all dialogues) but make it disabled for now.

**Comment by dannyparsons** [comment]
The button would lead to a subdialog with a large multiline textbox which displays the same comment as on the dialog just in way to more easily have a longer comment or multiple lines. The two textboxes should always display the same text.

**Comment by rdstern** [comment]
I assume there are 2 different points here.  First is the button, and I thought that Danny's idea of the ... continuation in a little box was nice and unobtrusive.  Then I don't have strong views on whether it simply opens a drop down window or goes to a text-box in a sub-dialogue.  I quite like the hint that is shown here.  With that information there is perhaps a slight point in having the sub-dialogue, because that information could be below the text-box and hence displayed all the time.

@dannyparsons may have stronger views?

**Comment by Patowhiz** [comment]
Okay. Let's go with button and the subdialog.. Should I put the button on the right of the current comment textbox. Any ideas on what would go to the subdialog apart from the multiline textbox? Should we have our subdialog base buttons included?

**Comment by rdstern** [comment]
Yes, I assume it will be just to the right of the current Comment field.

And it is just a simple text box there.  Possibly the simplest sub-dialogue in R-Instat!

Oh yes, of course we should have the base buttons.  Then the **Return** means we don't need anything clever like <ctrl> <Enter>.

**Comment by dannyparsons** [comment]
I actually like the dropdown button much better. This is more natural in this case than a subdialog, because it is only expanding the comment textbox - doesn't need other buttons or other options etc. like a subdialog. What we want to do here is the same as the example of Visual Studio.

But I don't have strong enough views if it's already been done in the other way.

Whichever way it is done, the bigger box is just an extension of the one of the main dialog, so they should always have the same text as each other (even though only one line is shown on the smaller one). 
We don't need to add # signs at the start of each line, this should be done when the script is run.

**Comment by rdstern** [comment]
OK.  If this is now done "the other way" and it would take some time to change, then I suggest we stick with the sub-dialogue - as |Danny says, he doesn't have strong-enough views.
On the other hand, I think you would have preferred your suggestion - as shown above.  And Danny says he likes the "dropdown button much more".  Not just "more", but "much more"!  So, if easy to go either way, then lets go to your idea!

**Comment by Patowhiz** [comment]
@dannyparsons @rdstern this is the design that I have finalised on in terms of our comment button position. I hope it's much better, at least that's what I thought, to have the button inside the textbox rather than outside. Below is an image that demonstrates this.

![comment button](https://user-images.githubusercontent.com/21179007/45418523-60793400-b68d-11e8-83ef-9a39aab73180.JPG)

**Comment by rdstern** [comment]
That's neat.  I like that.  @dannyparsons?

**Comment by Patowhiz** [comment]
This is the design I have finalised on. I chose building this to pop above rather than below because of the below space could be hidden especially of the parent dialog is big. The pop will be hidden when the user; presses Ctrl+Enter, clicks anywhere else, or clicks on the same ... button when its being shown(More of a toggle on off). 

<img width="326" alt="comment" src="https://user-images.githubusercontent.com/21179007/45422226-42fd9780-b698-11e8-97f9-ec43ec1d9440.png">

**Comment by rdstern** [comment]
Looks good.

**Comment by dannyparsons** [comment]
Looks great, thanks

**Comment by Patowhiz** [comment]
Well, this is all set now

**Comment by maxwellfundi** [comment]
@Patowhiz  what the state of this?

**Comment by rdstern** [comment]
What is the state now?  Can it be merged?
